renderdoc/driver/vulkan/wrappers/vk_device_funcs.cpp

```diff
@@ -229,6 +229,12 @@ ReplayStatus WrappedVulkan::Initialise(VkInitParams &params, uint64_t sectionVer
     }
   }
 
+// In development builds of RenderDoc, replay traces with validation
+// layers forced ON to see reported Vulkan VL errors.
+#if ENABLED(RDOC_DEVEL)
+  bool hasValidationLayers = false;
+#endif
```

Comment:
>> StripUnwantedLayers above removes the validation layers from replay
Yeah, I just wasn't sure why it's removed and if that could change in the future and I mostly wanted to avoid a situation where the standard VL is added a second time, because I've seen this cause hiccups in the loader. My first thought was to just change StripUnwantedLayers but I also didn't want to put this logic into the StripUnwantedLayers, because that would make the name of the function 

>> you also have to flip on the DISPLAY_RUNTIME_DEBUG_MESSAGES option
I'm not sure I flipped that, but I am certain I had the errors show up in the GUI. I can double check that. 

>>  validation layers have high perf impact as well as potentially adding crashes, and just as that's not desired in applications I don't want that in every RenderDoc development build
I can certainly relate to that... I can clearly see that this CL is not great and I can remove it from PR.

However, there is strong motivation to find a graceful way to allow enabling VL during replay, because in practice I've seen all vulkan apps have some sort of VK errors. Some of those apps would actually cause a crash during RD replay, and then the speculation is always that RD has a bug in replay code, although with my specific experience I've found that VL output was indicating an app bug, and when I would apply a fix for the error through code generator, RD replay would run as expected.

Reply:
Sure like I say I can understand the desire for it and it makes sense, but it needs to be opt-in. At the moment that means a compile-time switch because there's no way to do it at run-time. That can change if run-time replay options are implemented somehow.

The ideal would be if we could somehow do a pass to fetch the messages from a replay but then not have the layers active for the rest of the time, but no matter how I think about that it always entails loading the capture twice which would suck for a workflow.